app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java

```diff
@@ -1,45 +1,120 @@
 import android.os.Bundle;
+import android.os.Handler;
+import android.os.PowerManager;
+import android.os.Vibrator;
 import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
+import android.widget.TextView;
 
 import com.github.ppartisan.simplealarms.R;
 
-public final class AlarmLandingPageFragment extends Fragment implements View.OnClickListener {
 
+import static android.content.Context.VIBRATOR_SERVICE;
+
+public final class AlarmLandingPageFragment extends Fragment implements SensorEventListener {
+    public  long lastTime;
+    public  float speed;
+    public  float lastX;
+    public  float lastY;
+    public  float lastZ;
+    public  float x, y, z;
+    private int count;
+
+
+    public static  int SHAKE_THRESHOLD = 100;
+    public static  int DATA_X = SensorManager.DATA_X;
+    public static  int DATA_Y = SensorManager.DATA_Y;
+    public static  int DATA_Z = SensorManager.DATA_Z;
+    public SensorManager sensorManager;
+    public Sensor accelerormeterSensor;
+    private PowerManager pm;
+    private PowerManager.WakeLock wl;
+    private Vibrator vibrator;
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
 
         final View v = inflater.inflate(R.layout.fragment_alarm_landing_page, container, false);
-
+        vibrator = (Vibrator) getActivity().getSystemService(VIBRATOR_SERVICE);
+        sensorManager = (SensorManager) getActivity().getSystemService(Context.SENSOR_SERVICE);
+        accelerormeterSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
         final Button launchMainActivityBtn = (Button) v.findViewById(R.id.load_main_activity_btn);
         final Button dismiss = (Button) v.findViewById(R.id.dismiss_btn);
+        launchMainActivityBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(getContext(), MainActivity.class));
+                vibrator.cancel();
+                getActivity().finish();
+            }
+        });
+        dismiss.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                vibrator.cancel();
+                getActivity().finish();
+            }
+        });
+        pm=(PowerManager)getActivity().getSystemService(Context.POWER_SERVICE);
+        //wl=pm.newWakeLock(PowerManager.SCREEN_BRIGHT_WAKE_LOCK,"mytag");
 
-        launchMainActivityBtn.setOnClickListener(this);
-        dismiss.setOnClickListener(this);
 
+        startvibe();
         return v;
     }
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (sensorManager != null)
+            sensorManager.unregisterListener(this);
```

Comment:
Where do you set this listener?

Reply:
@PPartisan I added an api named 'sensorManager' and called 'unregisterListener' in it.
It is detailed in the link below.
https://developer.android.com/reference/android/hardware/SensorManager

Reply:
I understand - but don't you also need to call `sensorManager#registerListener` somewhere as well (presumably inside `onStart()`)?

Reply:
@PPartisan Ok! I understand! I added `onStart()`. You can turn off the alarm by shaking it. If you want, you can tweak some of features you mentioned (vibrating main activity, shake to dismiss, vibration pattern, number of shakes required to dismiss, etc.)